caampr-backend/cloudformation/template.yml

```diff
@@ -75,22 +75,26 @@ Resources:
     Type: AWS::DynamoDB::Table
     DeletionPolicy: Retain
     Properties:
-      TableName: Gear-v1
+      TableName: Gear-v2
       BillingMode: "PAY_PER_REQUEST"
       AttributeDefinitions:
         - AttributeName: "id"
           AttributeType: "S"
+        - AttributeName: "type"
+          AttributeType: "S"
         - AttributeName: "profileId"
           AttributeType: "S"
       KeySchema:
         - AttributeName: "id"
           KeyType: "HASH"
+        - AttributeName: "type"
+          KeyType: "RANGE"
       GlobalSecondaryIndexes:
         - IndexName: "gear-by-profile"
           KeySchema:
             - AttributeName: "profileId"
               KeyType: "HASH"
-            - AttributeName: "id"
+            - AttributeName: "type"
```

Comment:
Do we need to change the 2nd index range to "type" so we can filter it? Meaning, we couldn't filter based on gear type unless we set "type" to the range value?

Reply:
We could just grab all records based on the `profileId`, then do post-query filter in java.  Adding the `type` as the range key (sort key, whatever), dynamo will add it to the index itself, which allows it to efficiently query the `profileId` and then further filter based on `type` without scanning all items for the `profileId`.  Not a huge deal when there are a small number of records, but this can become a scaling problem, so better to plan now.